enclave-runner/src/usercalls/mod.rs

```diff
@@ -491,14 +503,51 @@ struct StoppedTcs {
 }
 
 struct IOHandlerInput<'tcs> {
-    tcs: &'tcs mut RunningTcs,
+    tcs: Option<&'tcs mut RunningTcs>,
     enclave: Arc<EnclaveState>,
     work_sender: &'tcs crossbeam::crossbeam_channel::Sender<Work>,
 }
 
+struct PendingEvents {
+    counts: [u32; Self::MAX_EVENT],
+}
```

Comment:
The impetus for this type is that some threads may never be interested in a particular event, and therefore we could be using unbounded amount of memory storing those events. For example most enclave threads are not interested in `EV_RETURNQ_NOT_EMPTY` since the return queue is probably owned by just one thread, but the enclave runner needs to send that event to all threads since it does not know which thread owns that queue.

Reply:
> The impetus for this type is that some threads may never be interested in a particular event, and therefore we could be using unbounded amount of memory storing those events.

Makes sense.

> the enclave runner needs to send [EV_RETURNQ_NOT_EMPTY] to all threads since it does not know which thread owns that queue

Well it only needs to be sent to all threads if no thread is currently waiting on that specific event:

> Userspace will wake any or all threads *waiting on the appropriate event* when it is triggered.

(emphasis mine)